modules/caddyhttp/headers/caddyfile.go

```diff
@@ -148,14 +174,36 @@ func parseReqHdrCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler,
 // will be used to search and then replacement will be used to
 // complete the substring replacement; in that case, any + or -
 // prefix to field will be ignored.
-func CaddyfileHeaderOp(ops *HeaderOps, field, value, replacement string) {
+func CaddyfileHeaderOp(ops *HeaderOps, field, value, replacement string) error {
+	return applyCaddyfileHeaderOp(ops, field, value, replacement, nil)
+}
+
+func applyCaddyfileHeaderOp(ops *HeaderOps, field, value, replacement string, respHeaderOps *RespHeaderOps) error {
 	if strings.HasPrefix(field, "+") {
 		if ops.Add == nil {
 			ops.Add = make(http.Header)
 		}
 		ops.Add.Set(field[1:], value)
 	} else if strings.HasPrefix(field, "-") {
 		ops.Delete = append(ops.Delete, field[1:])
+	} else if strings.HasPrefix(field, "?") {
+		if respHeaderOps == nil {
+			return fmt.Errorf("%v: %w", field, ErrBadOperation)
+		}
+
+		if respHeaderOps.Require == nil {
+			respHeaderOps.Require = &caddyhttp.ResponseMatcher{
+				Headers: make(http.Header, 1),
+			}
+		}
+		key := textproto.CanonicalMIMEHeaderKey(field[1:])
```

Comment:
Why is this needed? (The response matcher will do this already, see `getHeaderFieldVals()` in matchers.go.)